### DISCUSSION TITLE
Add Playwright to the defaults

### DISCUSSION THREAD

**Discussion by nl8io** [alagrede/znote-app#63] [Ideas]
It would be lovely to add Playwright to the defaults in order to run browser tests, automations, ... right from within zNote - and maybe even consider opening a "marketplace" for common scripts to (ab)use playwright for common tasks (e.g. scrape data from X and push into Y)

**Comment by alagrede** [comment]
Hello @nl8io,
I really like your suggestion on Playwright 😊. I'm currently working on a big refactoring to load external JS scripts. Maybe this update could bring something new to allow using libraries like Playwright.
I will study how to make this possible.

We keep in touch!
Anthony

**Comment by alagrede** [comment]
Hi @nl8io 👋,
Thank you again for your ideas! Not exactly what you want, because running test automation is not the main purpose of Znote, but I've added an example Playwright script to show how to manipulate a page with the library. This could be a good way to quickly prototype a playwright test or manipulation before writing the test in the right tool or project:
[Playwright script demo](https://github.com/alagrede/znote-app/blob/master/examples/playwright-script-demo.md)

I hope this will help you.
Regards,
Anthony

**Reply by nl8io** [reply]
Thats awesome and "good enough". However, I am (ab)using playwright a lot amongst Browser Testing but for automating simplpe tasks I got (recording them is so easy and playing them back even more - compared to any browser automation suite I saw + its free.

So integrating it natively, would have just been something that, I thought, could  also open new use cases for you, since I am confident I am not the only one using playwright that way.

Thanks for the example shared. Honestly appreciate it.

**Reply by alagrede** [reply]
I get your point about automating tasks and you're right, using Playwright as an automation tool with Znote could be very powerful 😊 
I will continue to investigate how to do an effective integration with Playwright and come back with concrete use cases.
Happy to help you.
Anthony
